File: BOJ/10828.cpp
#include<bits/stdc++.h>
#define endl "\n"
using namespace std;
int N;
stack<int> s;
queue<int> q;
void stk_cmd()
{
    string input;
    cin >> input;
    int cmd;
    if(input=="push")
    {
        int tmp; cin >> tmp;
        s.push(tmp);
    }
    else if(input=="pop")
    {
        if(s.empty())
        {
            q.push(-1);
            return;
        }
        q.push(s.top()); 
        s.pop();
    }
    else if(input=="top")
    {
        if(s.size()==0)
        {
            q.push(-1);
            return;
        }
        q.push(s.top());
    }
    else if(input =="size")
    {
        q.push(s.size());
    }
    else
    {
        q.push(s.empty());
    }
}

int main()
{
    cin >> N;
    for(int i=0;i<N;i++)
    stk_cmd();

    while(!q.empty())
    {
        cout << q.front()<<endl; q.pop();
    }
}

File: BOJ/10828.py
import sys

input=sys.stdin.readline

def main() :
    stk = []
    N = int(input())
    for i in range(N) :
        action = input().strip()
        if action =='pop' :
            if not stk : 
                print(-1)
            else : 
                print(stk.pop())
        elif action == 'size' : 
            print(len(stk))
        elif action == "empty" :
            if(not stk) :
                print(1)
            else :
                print(0)
        elif action == "top" :
            if(not stk) : 
                print(-1)
            else : print(stk[-1])
        else :
            num = int(action.split()[1])
            stk.append(num)

main()